diff -Nru libjsw-1.5.5.orig/jscalibrator/disk.cpp libjsw-1.5.5/jscalibrator/disk.cpp
--- libjsw-1.5.5.orig/jscalibrator/disk.cpp	2004-12-21 06:55:58.000000000 +0100
+++ libjsw-1.5.5/jscalibrator/disk.cpp	2005-05-15 20:00:36.829988120 +0200
@@ -132,7 +132,7 @@
 
 	// Dan S: typecast is due to const.
 #if defined(__cplusplus) || defined(c_plusplus)
-	while(ISBLANK(reinterpret_cast<char>(*path)))
+	while(ISBLANK(*path))
 #else
 	while(ISBLANK(*path))
 #endif
diff -Nru libjsw-1.5.5.orig/libjsw/disk.cpp libjsw-1.5.5/libjsw/disk.cpp
--- libjsw-1.5.5.orig/libjsw/disk.cpp	2004-12-21 06:55:57.000000000 +0100
+++ libjsw-1.5.5/libjsw/disk.cpp	2005-05-15 19:59:45.254828736 +0200
@@ -132,7 +132,7 @@
 
 	// Dan S: typecast is due to const.
 #if defined(__cplusplus) || defined(c_plusplus)
-	while(ISBLANK(reinterpret_cast<char>(*path)))
+	while(ISBLANK(*path))
 #else
 	while(ISBLANK(*path))
 #endif
