--- libjsw-1.5.6/jscalibrator/Makefile.orig	2006-03-31 08:28:01.000000000 +0200
+++ libjsw-1.5.6/jscalibrator/Makefile	2006-04-23 18:55:57.353808750 +0200
@@ -85,7 +85,7 @@
 #   Each argument is of the format -L<dir> where <dir> is the full
 #   path to the directory.
 #
-LIB_DIRS =
+LIB_DIRS = -L ../libjsw
 
 # Header File Directories:
 #
@@ -131,7 +131,7 @@
 
 modules: $(OBJ_C) $(OBJ_CPP)
 	@echo  -n "Linking modules..."
-	@$(CC) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIBS) $(LIB_DIRS)
+	@$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIB_DIRS) $(LIBS)
 	@echo -n "   "
 	@-$(LS) $(LSFLAGS) $(BIN)
 
--- libjsw-1.5.6/libjsw/Makefile.orig	2006-03-31 08:28:00.000000000 +0200
+++ libjsw-1.5.6/libjsw/Makefile	2006-04-23 19:04:41.278552000 +0200
@@ -96,7 +96,7 @@
 
 modules: $(OBJ_C) $(OBJ_CPP)
 	@echo  -n "Linking modules..."
-	@$(CC) $(OBJ_C) $(OBJ_CPP) -Wl,-soname=$(LIB) -shared -o $(LIB) $(LIBS) $(LIB_DIRS)
+	@$(CPP) $(OBJ_C) $(OBJ_CPP) -Wl,-soname=$(LIB) -shared -o $(LIB) $(LIB_DIRS) $(LIBS)
 	@echo -n "   "
 	@$(RM) $(RMFLAGS) $(LIBPFX).so
 	@$(LINK) -s $(LIB) $(LIBPFX).so
