--- libjsw-1.5.2/include/xsw_ctype.h.orig	2003-04-30 23:49:39.000000000 +0200
+++ libjsw-1.5.2/include/xsw_ctype.h	2003-05-10 01:01:08.000000000 +0200
@@ -1,7 +1,7 @@
 // xsw_ctype.h
 // This is intended as a prototype for files using the global/ctype.cpp file.
 
-#if !defined(__FreeBSD__) && !defined(__NetBSD__)
+#if !defined(__FreeBSD__) && !defined(__NetBSD__) && !defined(linux)
 # if defined(__cplusplus) || defined(c_plusplus)
 
 /* Most systems should have this defined.
