--- libjsw-1.5.2/jscalibrator/Makefile.orig	Tue Nov  6 00:55:10 2001
+++ libjsw-1.5.2/jscalibrator/Makefile	Sat Feb  1 22:37:20 2003
@@ -85,7 +85,7 @@
 #   Each argument is of the format -L<dir> where <dir> is the full
 #   path to the directory.
 #
-LIB_DIRS =
+LIB_DIRS = -L ../libjsw
 
 # Header File Directories:
 #
@@ -121,7 +121,7 @@
 
 modules: $(OBJ_C) $(OBJ_CPP)
 	@echo "Linking modules..."
-	@$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIBS) $(LIB_DIRS)
+	@$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIB_DIRS) $(LIBS)
 
 prebuild:
 	@echo "Building program \"$(BIN)\"..."
--- libjsw-1.5.2/libjsw/Makefile.orig	Tue Nov  6 00:55:10 2001
+++ libjsw-1.5.2/libjsw/Makefile	Sat Feb  1 22:39:52 2003
@@ -87,7 +87,7 @@
 
 modules: $(OBJ_C) $(OBJ_CPP)
 	@echo "Linking modules..."
-	@$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(LIB) $(LIBS) $(LIB_DIRS)
+	@$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(LIB) $(LIB_DIRS) $(LIBS) -Wl,-soname=$(LIBPFX).so.1.5
 
 prebuild:
 	@echo "Building library \"$(LIB)\"..."
