--- libjsw-1.5.5/libjsw/string.cpp.orig	2004-12-21 06:55:57.000000000 +0100
+++ libjsw-1.5.5/libjsw/string.cpp	2004-12-27 00:56:56.800834104 +0100
@@ -28,7 +28,6 @@
 #ifdef __MSW__
 int strcasecmp(const char *s1, const char *s2);
 #endif
-char *strcasestr(const char *haystack, const char *needle);
 int strpfx(const char *s, const char *pfx);
 int strcasepfx(const char *s, const char *pfx);
 
@@ -213,6 +212,7 @@
 }
 #endif
 
+#ifndef __linux__ /* glibc */
 /*
  *	Case insensitive version of strstr(). Returns the pointer to
  *	needle in haystack if found or NULL on no match.
@@ -274,6 +274,7 @@
 
 	return(NULL);
 }
+#endif
 
 /*
  *	Checks if string pfx is a prefix of string s.
--- libjsw-1.5.5/include/string.h.orig	2004-12-21 06:55:58.000000000 +0100
+++ libjsw-1.5.5/include/string.h	2004-12-27 00:57:04.965592872 +0100
@@ -43,7 +43,9 @@
 #ifdef __MSW__
 extern int strcasecmp(const char *s1, const char *s2);
 #endif
+#ifndef __linux__ /* glibc */
 extern char *strcasestr(const char *haystack, const char *needle);
+#endif
 extern int strpfx(const char *s, const char *pfx);
 extern int strcasepfx(const char *s, const char *pfx);
 
--- libjsw-1.5.5/jscalibrator/string.cpp.orig	2004-12-21 06:55:58.000000000 +0100
+++ libjsw-1.5.5/jscalibrator/string.cpp	2004-12-27 01:11:07.256545184 +0100
@@ -28,7 +28,6 @@
 #ifdef __MSW__
 int strcasecmp(const char *s1, const char *s2);
 #endif
-char *strcasestr(const char *haystack, const char *needle);
 int strpfx(const char *s, const char *pfx);
 int strcasepfx(const char *s, const char *pfx);
 
@@ -213,6 +212,7 @@
 }
 #endif
 
+#ifndef __linux__ /* glibc */
 /*
  *	Case insensitive version of strstr(). Returns the pointer to
  *	needle in haystack if found or NULL on no match.
@@ -274,6 +274,7 @@
 
 	return(NULL);
 }
+#endif
 
 /*
  *	Checks if string pfx is a prefix of string s.
